### PR TITLE
add basic api call

### PR DESCRIPTION
**Pull request by vitalikdanchenko**

### CODE REVIEW

File: lib/talkable/api/base.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        end
+
+        protected
+
+        def request_params(params = nil)
+          (params || {}).merge({
+            api_key:   Talkable.configuration.api_key,
+            site_slug: Talkable.configuration.site_slug,
+          })
+        end
```
Comment (Tonkonozhenko) [inline lib/talkable/api/base.rb:29]:
``` ruby
        def request_params(params = {})
           params.merge(
             api_key:   Talkable.configuration.api_key,
             site_slug: Talkable.configuration.site_slug,
           )
         end
```

File: lib/talkable/api/base.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+          }
+        end
+
+        def perform_request(uri, request)
+          http = Net::HTTP.new(uri.host, uri.port)
+          http.use_ssl = uri.is_a?(URI::HTTPS)
+
+          request.initialize_http_header request_headers
+          process_response http.request(request)
+        rescue  Errno::ECONNREFUSED, Errno::ECONNRESET, Errno::ETIMEDOUT,
```
Comment (Tonkonozhenko) [inline lib/talkable/api/base.rb:53]: I think we need to move all this classes to constant

Comment (vitalikdanchenko) [inline lib/talkable/api/base.rb:53]: I don't see any place where I can reuse it so far

File: lib/talkable/api/base.rb [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+          when Net::HTTPSuccess, Net::HTTPClientError
+            parse_response response.body
+          when Net::HTTPServerError
+            raise NetworkError.new("Server #{Talkable.configuration.server} is unavailable. Try again later")
+          end
+        end
+
+        def parse_response(body)
+          raise_invalid_response if body.nil?
+          result = JSON.parse(body)
```
Comment (Tonkonozhenko) [inline lib/talkable/api/base.rb]: What about to add `symbolize_names` param here?

File: lib/talkable/api/base.rb [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+          raise_invalid_response unless result.is_a?(Hash)
```
Comment (Tonkonozhenko) [inline lib/talkable/api/base.rb:71]: Do you think this situation can happen?

File: lib/talkable/api/offer.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+module Talkable
+  module API
+    class Offer < Base
+      class << self
+        def find(id, params = nil)
```
Comment (Tonkonozhenko) [inline lib/talkable/api/offer.rb]: I suppose `nil` will cause an exception [here](https://github.com/talkable/talkable-ruby/pull/2/files#diff-20f9c518bcb040c78c08bda770478f18R25)

File: spec/api/base_spec.rb [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    let(:api_key)     { 'api_key' }
+    let(:version)     { Talkable::API::VERSION }
+    let(:http_method) { http_method }
+    let(:path)        { '/path' }
+    let(:params)      { {key1: :value1, key2: :value2} }
+
+    def stub_api_request
+      api_params = params.merge({api_key: api_key, site_slug: site_slug})
+      stub_request(http_method, "#{server}/api/#{version}/path").
+        with(http_method == :get ? {query: api_params} : {body: api_params.to_json})
```
Comment (Tonkonozhenko) [inline spec/api/base_spec.rb]: What about to move this to `.get` and `.post` contexts?

File: lib/talkable/api/base.rb (hunk 3 above)
Comment (vitalikdanchenko) [inline lib/talkable/api/base.rb]: agree

**Comment by Tonkonozhenko** [comment]
Will we spec working with real API?

File: lib/talkable/api/base.rb (hunk 4 above)
Comment (vitalikdanchenko) [inline lib/talkable/api/base.rb:71]: I'd better be sure that we would caught it if it happened

**Comment by vitalikdanchenko** [comment]
no, there is no need of that

File: spec/api/base_spec.rb (hunk 6 above)
Comment (vitalikdanchenko) [inline spec/api/base_spec.rb]: how do you see this?

Comment (Tonkonozhenko) [inline spec/api/base_spec.rb]:
``` ruby
...

let(:api_params) { params.merge({api_key: api_key, site_slug: site_slug}) }
let!(:stub_api_request) { stub_request(http_method, "#{server}/api/#{version}/path").with(request_stubs) }

...

context '.get' do
  let(:request_stubs) { {query: api_params} }

  ...
end

context '.post' do
  let(:request_stubs) { {body: api_params.to_json} }

  ...
end
```

**Comment by Tonkonozhenko** [comment]
LGTM

**Pull request merged**